Evaluation-2/src/expression-test.js

```diff
@@ -1,127 +1,92 @@
-import { createStack } from "./stack.js"
-export function expressionTest(){
-let stack=createStack()
-return{
-    checkExpr(expr){
-        let str=expr;
-        let expArr=[]
-        for(let i=0;i<str.length;i++){
-            if(str[i]=='1'||str[i]=='2'||str[i]=='3'||str[i]=='4'||str[i]=='5'||str[i]=='6'||str[i]=='7'||str[i]=='8'||str[i]=='9'||str[i]=='+'||str[i]=='-'){
-
-                expArr.push(str[i])
-            }
+/**
+ * function takes an array containing expression 
+ * @param {*} arr computes the array and returns it
+ * @returns a computed or evaluated expression
+ */
+function compute(arr){
+    let operandSt=[]
+    let operatorSt=[]
+    let result=[]
+    for(let a of arr){
+        if(a=='0'||a=='1'||a=='2'||a=='3'||a=='4'||a=='5'||a=='6'||a=='7'||a=='8'||a=='9'){
+            operandSt.push(a)
         }
-        console.log(expArr)
-        let exp1=[]
-        let exp2=[]
-        let exp3=[]
-        for(let i=0;i<=4;i++){
-            exp1.push(expArr[i])
+        else{
+            operatorSt.push(a)
         }
-        for(let i=5;i<=7;i++){
-            exp2.push(expArr[i])
+        
+        
+    }
+        let resultArr=[]
+        while(operatorSt.length>0){
+        let op1=operandSt.shift()
+        let op2=operandSt.shift()
+         if (op1 !== undefined && op2 !== undefined) {
+             let o1=parseFloat(op1)
+             let o2=parseFloat(op2)
+        // console.log('operator1 = '+o1+' Operator 2= '+o2)
+        let operator=operatorSt.shift()
+        if(operator==='+'){
+            operandSt.unshift(o1+o2)
         }
-        for(let i=8;i<expArr.length;i++){
-            exp3.push(expArr[i])
+        else if(operator==='-'){
+            operandSt.unshift(o1-o2)
         }
-        for(let i=exp1.length-1;i>0;i--){
-        stack.push(exp1[i])
         }
-        // console.log("Stack = ",stack.peek())
-        console.log(exp1," ",exp2," ",exp3)
-        let str1=0
-        str1=Number(exp1[0])
-        stack.pop()
-        for(let i=1;i<exp1.length;i++){
-            
-            if(exp1[i]==='+'||exp1[i]==='-'){
-                if(exp1[i]=='+'){
-                    stack.pop()
-                    str1=str1+Number(exp1[i+1])
-                    stack.pop()
-                }
-                else{
-                    if(stack.peek()=='-'){
-                        stack.pop()
-                        str1=str1-Number(exp1[i+1])
-                        stack.pop()
-                    }
-                }
-            }
-            else if(!(exp1[i]=='+'||exp1[i]=='-')){
-                continue
-            }
-            
-            }
-            if(exp1.length>3){
-                str1=str1-exp1[2]
+        
+        }
+        return operandSt
+}
+/**
+ * Function takes a string
```

Comment:
Wrong description and poor naming of this function. Function names need to be verbs. like in this case `evaluateExpression`

Reply:
fixed it